mmocr/models/textrecog/backbones/svtr.py

```diff
@@ -0,0 +1,650 @@
+            kernel_size=local_k,
+            stride=1,
+            padding=(local_k[0] // 2, local_k[1] // 2),
+            groups=num_heads)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward function.
+
+        Args:
+            x (torch.Tensor): A Tensor of shape :math:`(N, HW, C)`.
+
+        Returns:
+            torch.Tensor: Tensor: A tensor of shape math:`(N, HW, C)`.
+        """
+        h, w = self.HW
+        x = x.permute(0, 2, 1).reshape([-1, self.embed_dims, h, w])
+        x = self.local_mixer(x)
+        x = x.flatten(2).permute(0, 2, 1)
+        return x
+
+
+class AttnMixer(BaseModule):
+    """One of mixer of {'Global', 'Local'}. Defaults to Global Mixer.
+
+    Args:
+        embed_dims (int): Number of character components.
+        num_heads (int, optional): Number of heads. Defaults to 8.
+        mixer (str, optional): The mixer type. Defaults to 'Global'.
+        HW (Tuple[int, int], optional): Number of H x W. Defaults to [8, 25].
+        local_k (Tuple[int, int], optional): Window size. Defaults to [7, 11].
+        qkv_bias (bool, optional): Whether a additive bias is required.
+            Defaults to False.
+        qk_scale (float, optional): A scaling factor. Defaults to None.
+        attn_drop (float, optional): A Dropout layer. Defaults to 0.0.
+        proj_drop (float, optional): A Dropout layer. Defaults to 0.0.
+        init_cfg (dict or list[dict], optional): Initialization configs.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 embed_dims: int,
+                 num_heads: int = 8,
+                 mixer: str = 'Global',
+                 HW: Tuple[int, int] = [8, 25],
+                 local_k: Tuple[int, int] = [7, 11],
+                 qkv_bias: bool = False,
+                 qk_scale: float = None,
+                 attn_drop: float = 0.,
+                 proj_drop: float = 0.,
+                 init_cfg: Optional[Union[Dict, List[Dict]]] = None):
+        super().__init__(init_cfg)
+        assert mixer in {'Global', 'Local'}, \
+            "The type of mixer must belong to {'Global', 'Local'}"
+        self.num_heads = num_heads
+        head_dim = embed_dims // num_heads
+        self.scale = qk_scale or head_dim**-0.5
+        self.qkv = nn.Linear(embed_dims, embed_dims * 3, bias=qkv_bias)
+        self.attn_drop = nn.Dropout(attn_drop)
+        self.proj = nn.Linear(embed_dims, embed_dims)
+        self.proj_drop = nn.Dropout(proj_drop)
+        self.HW = HW
+        if HW is not None:
+            H, W = HW
+            self.N = H * W
+            self.C = embed_dims
+        if mixer == 'Local' and HW is not None:
+            hk = local_k[0]
+            wk = local_k[1]
+            mask = torch.ones([H * W, H + hk - 1, W + wk - 1],
+                              dtype=torch.float32)
+            for h in range(0, H):
+                for w in range(0, W):
+                    mask[h * w + w, h:h + hk, w:w + wk] = 0.
```

Comment:
```suggestion
                    mask[h * W + w, h:h + hk, w:w + wk] = 0.
```
Try to revise these variable names to avoid confusion